fastpath/fastpath/core.py

```diff
@@ -1622,25 +1622,20 @@ def write_measurement_to_disk(msm_tup) -> None:
         msm_tup: Measurement tuple as it comes from the work queue
     """
 
+    # msmt_uid is a unique id based on upload time, cc, testname and hash
     data, measurement, msmt_uid = msm_tup
     ts, cc, test_name, h = msmt_uid.split("_")
 
-    now = datetime.now(timezone.utc)
+    now = datetime.strptime(ts, "%Y%m%d%H%M%S.%f")
```

Comment:
I think you can just extract the `hour` value directly from the `ts` string by extracting the first 8 characters, there is no need to go through the datetime parser.

I suggest doing that to keep it simpler.

Reply:
Wouldn't that be the first 10 characters? 
Example: 
```
>>> now.strftime("%Y%m%d%H")
'2025080813'
>>> len(now.strftime("%Y%m%d%H"))
10
```